.github/workflows/build-docker-image.yml

```diff
@@ -0,0 +1,38 @@
+name: Build Docker Images
+on:
+  push:
+    tags:
+      - v*
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        bin: 
+          - ssserver
+          - sslocal
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1 
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Docker metadata
+        id: metadata
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/${{ github.repository_owner }}/${{ matrix.bin }}-rust
```

Comment:
Is the postfix `-rust` really necessary? Could we tag the image with name like `ghcr.io/shadowsocks-rust/sslocal`?

Reply:
Since the full name format of the GCR image is `ghcr.io/OWNER/IMAGE_NAME:latest`, images belong to OWNER not repository, they are just bound to this repository, so the OWNER can only be `shadowsocks`. I added the rust suffix to prevent conflicts with the naming of [shadowsocks-libev](https://github.com/shadowsocks/shadowsocks-libev)'s sserver and sslocal.

This is a limitation of the GCR and I have no way to modify it.